extensions/npm/src/tasks.ts

```diff
@@ -155,6 +155,28 @@ async function detectNpmScripts(): Promise<Task[]> {
 	}
 }
 
+
+export async function detectNpmScriptsForFolder(folder: string): Promise<Task[]> {
```

Comment:
There already is a cache of detected npm scripts, rather then recomputing the scripts for a folder I suggest to look into whether the existing Task cache can be leveraged https://github.com/microsoft/vscode/blob/6e8ff27ea624175da47d0b02282774c58644b978/extensions/npm/src/tasks.ts#L25, e.g., by iterating over the cached tasks and checking whether an item is contained in the folder.